libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -96,6 +110,15 @@ func New(cfg *common.Config) (processors.Processor, error) {
 		return nil, err
 	}
 
+	if !isKubernetesAvailable(client) {
+		return &kubernetesAnnotator{
```

Comment:
Ok, I will along with this! I like it. Thank you for suggesting!

I initialize `kubernetesAvailable` by default to `true` and set it to false inside the if-branch to make it more clear, I think there is no need to implicitly set to `true` inside the main flow, just use the initialized one.

Reply:
cool

This is very opinionated but, here I go:
My problem with setting default to `true` vs leaving it blank is that code stop being self explanatory. The default is unknown, from my POV leaving it to the type's default is ok since we don't know the value yet.

Then once you call `isKubernetesAvailable` it is ok to set it to `false` and return if it is not available, or set it to `true` explicitly if it is available.
For this case the difference is minimal, a glimpse of the code will tell you what it is doing, but I still find it better to set vars at the most meaningful place in code.

Reply:
All right, I got your point! I will go along with you.